src/lib/parser.rs

```diff
@@ -246,21 +277,45 @@ impl<'a, TokId: Clone + Copy + Debug + PartialEq + TryFrom<usize> + TryInto<usiz
     }
 }
 
+pub enum RecoveryKind {
+    Corchuelo,
+    KimYi
+}
+
 /// Parse the lexemes, returning either a parse tree or a vector of `ParseError`s.
 pub fn parse<TokId: Copy + Debug + PartialEq + TryFrom<usize> + TryInto<usize>>
-            (grm: &YaccGrammar, sgraph: &StateGraph, stable: &StateTable,
-             lexemes: &Vec<Lexeme<TokId>>)
-         -> Result<Node<TokId>, (Node<TokId>, Vec<ParseError<TokId>>)>
+       (grm: &YaccGrammar, sgraph: &StateGraph, stable: &StateTable,
+        lexemes: &Vec<Lexeme<TokId>>)
+    -> Result<Node<TokId>, (Node<TokId>, Vec<ParseError<TokId>>)>
+{
+    parse_rcvry(RecoveryKind::KimYi, grm, |_| 1, |_| 1, sgraph, stable, lexemes)
+}
+
+/// Parse the lexemes, specifying a particularly type of error recovery, returning either a parse
+/// tree or a vector of `ParseError`s.
+pub fn parse_rcvry
```

Comment:
Hah, you really couldn't be bothered to type those 3 extra characters? ;-)

Reply:
Does it? I found it easier to type `recovery` than `rcvry`.

Reply:
I can change it if you think it would be better.

Reply:
It's not the end of the world, but I always find it easier to type full words rather than abbreviations that force me to think about how it was typed. Then again, most editors have autocomplete.